sdk/core/Azure.Core.Testing/src/Azure.Core.Testing.csproj

```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>A set of testing primitives for libraries that use Azure.Core</Description>
+    <AssemblyTitle>Microsoft Azure Client Pipeline Testing</AssemblyTitle>
+    <Version>1.0.0-preview.5</Version>
```

Comment:
The version will not be interesting unless we decide to release a package which I'd prefer to wait on making that decision as we talked lets keep trying P2P, with a conditionalized Package or Project reference to Azure.Core, and if that doesn't end up working out then we can consider shipping a package.

Reply:
I don't think the conditionalized reference would work here. `Azure.Core.Tests` project uses test framework project so it has to be P2P unconditionally.

Reply:
Just so it is clear what we are talking about:

Azure.Core.Tests ->P2P Azure.Core.Testing -> P2P Azure.Core

Azure.ApplicationModel.Configurtation.Tests -> P2P Azure.Core.Testing -> PackageReference Azure.Core 

I believe that we can make that work as long as we keep the set of public APIs Azure.Core.Testings is using stable and even if we do make usages of some new APIs we could consider #ifdef for the delta temporarily.

Reply:
```
Azure.Core.Tests ->P2P Azure.Core.Testing -> P2P Azure.Core

Azure.ApplicationModel.Configurtation.Tests -> P2P Azure.Core.Testing -> PackageReference Azure.Core

```

What condition are we going to use for `Azure.Core.Testing -> Azure.Core` reference? I don't see how it would work in `dotnet build data-plane.proj` where all projects would be unified into one closure.

Reply:
You can pass a unique global property from the Azure.Core.Tests to Azure.Core.Testing ProjectReference which would essentially cause msbuild to build that Azure.Core.Testing library twice and in the one case it would do a ProjectReference to Azure.Core and in the other case it would do a PackageReference. Of course we would also need to update the output paths with this config information to ensure that the builds don't stomp on each other.

Reply:
It would work to build the project multiple times, not sure if nuget and VS would be as happy. In general, seems slightly complicated considering what we are trying to accomplish. Switching between Azure.Core.Testing P2P/ProjectReference based on the same logic we use for Azure.Core feels way easier.

Reply:
We could also just have shared source code that gets directly included into the test projects.